openspoor/visualisations/trackmap.py

```diff
@@ -79,82 +89,71 @@ def _add_luchtfoto(self) -> None:
+        Set the zoom level so all the plotted objects fit neatly within the initial view of the output.
 
         :return: None
         """
         bboxes = []
 
         for _, item in self._children.items():
-            for _, subitem in item._children.items():
-                # This is for plotted linestrings
-                if isinstance(subitem.__dict__['_parent'], folium.features.Choropleth):
-                    chloropleth = subitem.__dict__['_parent']
-                    chloropleth_features = chloropleth.__dict__['geojson'].__dict__['data']['features']
-                    for feature in chloropleth_features:
-                        bboxes.append(
-                            feature['bbox'][::-1])  # Reverse lat and long ordering here
-                try:
-                    # This is for plotted markers
-                    bboxes.append(
-                        subitem.__dict__['data']['features'][0]['bbox'][::-1])  # Reverse lat and long ordering here
-                except:
-                    pass
-            try:
-                bboxes.append(item.location * 2)  # Change order
-            except:
-                pass
+            if isinstance(item, folium.features.Choropleth):  # For linestrings
+                bboxes.extend(feature['bbox'][::-1] for feature in item.geojson.data['features'])
+            if isinstance(item, folium.features.GeoJson):  # For areas
+                bboxes.append([i for coords in item.get_bounds() for i in coords])
+            if isinstance(item, folium.map.Marker):  # For markers
+                bboxes.append(item.location * 2)  # Min and max bound are equal for points, hence the repeat
 
         if bboxes:  # Fit only if there are some items to show
             bounds = [min(map(lambda x: x[i], bboxes)) if i < 2 else max(map(lambda x: x[i], bboxes)) for i in range(4)]
             self.fit_bounds([bounds[:2], bounds[2:]])
 
-    def add(self, plotobject):
+    def add(self, plotobject) -> TrackMap:
         plotobject.add_to(self)
         return self
 
-    def show(self, makesmaller: bool = False):
+    def show(self, notebook: bool = False) -> Union[TrackMap, folium.Figure]:
         """
-        Fix the map so it is zoomed to a nice level and is displayed at a suitable size.
+        Show the map zoomed to a nice level and displayed at a suitable size.
 
-        :param makesmaller: A boolean for decreasing the size of the final plot. Designed for Databricks Notebooks
+        :param notebook: A boolean for decreasing the size of the final plot. Designed for use in Notebooks
         :return:
         """
         self._fix_zoom()
-        if makesmaller:
-            self.superfigure = folium.Figure(1000, 400)
-            self.add_to(self.superfigure)
-            return self.superfigure
+        if notebook:
+            figure = folium.Figure(1000, 400)
+            self.add_to(figure)
```

Comment:
Trackmap does not yet contain an add_to method, only implemented in ABC.

Reply:
This uses the add_to functionality in folium.Map.

Reply:
Ah, maybe a bit confusing with the add_to function in this file

Reply:
I've added an inline comment to explain the why here.